agent/api/task/service_connect.go

```diff
@@ -65,3 +80,55 @@ type DNSConfigEntry struct {
 	HostName string `json:"hostName"`
 	Address  string `json:"address"`
 }
+
+// ExtractSCConfigFromACSAttachment finds the service connect attachment, and extracts the service connect container name and service connect config value.
+func ExtractSCConfigFromACSAttachment(acsTask *ecsacs.Task) (*ServiceConnectConfig, error) {
+	if acsTask.Attachments == nil {
+		return nil, nil
+	}
+	var scAttachment *ecsacs.Attachment
+	containerName := ""
+	for _, attachment := range acsTask.Attachments {
+		if aws.StringValue(attachment.AttachmentType) == serviceConnectAttType {
+			scAttachment = attachment
```

Comment:
Shouldnt this be done at the task level where we iterate on all attachments and dependin on the type of attachment, we parse each of them? So, we will have `parseServiceConnectAttachment` within `service_connect.go` being called from `task.go`?

Reply:
Thanks for the suggestion. Logic to get the service connect attachment is moved to task.go, and the function is renamed to `ParseServiceConnectAttachment` in commit 8e69368bc5c1d7bd3ab71391103f4b6a69f06054.